pkg/apis/istio/v1alpha2/deepcopy.go

```diff
@@ -15,13 +15,22 @@
 package v1alpha2
 
 import (
-	runtime "k8s.io/apimachinery/pkg/runtime"
-	"reflect"
+	"github.com/prometheus/common/log"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"istio.io/operator/pkg/util"
 )
 
 // DeepCopyInto is an autogenerated deepcopy function, copying the receiver, writing into out. in must be non-nil.
 func (in *IstioControlPlane) DeepCopyInto(out *IstioControlPlane) {
-	reflect.ValueOf(out).Elem().Set(reflect.ValueOf(in).Elem())
+	inj, err := util.MarshalWithJSONPB(in)
```

Comment:
There's a better to write/generate deepcopy functions.

You should use `proto.Clone()` to do a correct deepcopy at the top level proto message.

Here's an example:
```go
package v1alpha1

import (
	"github.com/gogo/protobuf/proto"

)

// DeepCopyInto supports using Application within kubernetes types, where deepcopy-gen is used.
func (in *IstioControlPlane) DeepCopyInto(out *IstioControlPlane) {
	*out = *proto.Clone(in).(*IstioControlPlane)
}
```

We also have tools to auto-generate all deepcopy functions for you if you don't want to write this manually.

@ostromart fyi

Reply:
Thanks Neeraj for the proto deepcopy tip, I'm making the change you suggested. With the auto-generation, the issue I ran into a while back is that eventually you hit k8s types that don't have DeepCopyInto implemented and then you have to somehow generate zz_deepcopy files for those (in the vendor tree). I could not figure out a clean way to do it but if you know how it would be great to have.